required/latex-lab/latex-lab-block.dtx

```diff
@@ -3971,15 +4092,38 @@
 %    Finally, at the list end we have to close the open
 %    \texttt{LBody}, \texttt{LI}, \texttt{L}, and possibly a
 %    \struct{text} if the last item ends with a list.
+%    However, if the user forgot to add an \cs{item} there is no
+%    \texttt{LI} and \texttt{LBody} open, so we check for the status
+%    of \texttt{@newlist}. The corresponding no-item error was
+%    generated earlier outside the tagging code.
+%
+%    One could argue that it doesn't matter if the tagging is wrong
+%    after a \cs{@noitemerr} was issued. However, there is one case
+%    where it isn't an error: In the \texttt{thebibliography}
+%    environment (which is internally a list) it is often the case
+%    that documents start out with an empty environment, not
+%    containing any \cs{bibitem}s. For that reason \cs{@noitemerr} is
+%    redefined inside that environment to only produce a warning;
+%    hence we have to produce correct tag structures in that case.
+%  \changes{v0.8q}{2024/09/02}{Do not close LI and LBody if they
+%    never were opened.}
 %    \begin{macrocode}
 \cs_set:Npn \@@_list_end: {
-  \legacy_if:nT { @endpe }
+%    \end{macrocode}
+%    If \texttt{@newlist} is true (ie when we have an error or warning
+%    situation) there is not much to close. If \verb=<L></L>= is not
+%    valid syntax we could add an empty \verb=<LI>= at this point instead.\fmi{check}
```

Comment:
Info: Currently it is clearly stated that any PDF structure can be empty, but this is a question that is under active, often heated, discussion!